panda/src/components/ProductGallery.js

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import "./ProductGallery.css";
+import heart from "../icon/ic_heart.png";
+
+function ProductGallery() {
+  const [products, setProducts] = useState([]);
+  const [error, setError] = useState(null);
+  const [searchQuery, setSearchQuery] = useState("");  // searchQuery 상태 정의
+  const [sortOption, setSortOption] = useState("latest");  // sortOption 상태 정의
+
+  useEffect(() => {
+    // API에서 limit과 offset 파라미터를 사용하여 데이터를 가져오기
+    axios
+      .get("https://panda-market-api.vercel.app/products", {
+        params: {
+          limit: 14, // 한 번에 가져올 상품 수
+          offset: 0, // 처음부터 14개 상품을 가져옴
+        },
+      })
+      .then((response) => {
+        console.log(response.data); // 응답 데이터 확인
+        if (response.data && Array.isArray(response.data.list)) {
+          setProducts(response.data.list); // 상품 데이터를 설정
+        } else {
+          setError("상품 데이터를 불러오는 데 문제가 발생했습니다.");
+        }
+      })
+      .catch((error) => {
+        console.error("상품 데이터를 불러오는 데 문제가 발생했습니다:", error);
+        setError("상품 데이터를 불러오는 데 문제가 발생했습니다.");
+      });
+  }, []);
+
+  // 좋아요 증가 함수
+  const handleLike = (productId) => {
+    setProducts((prevProducts) =>
+      prevProducts.map((product) =>
+        product.id === productId
+          ? { ...product, favoriteCount: product.favoriteCount + 1 }
+          : product
+      )
+    );
+  };
```

Comment:
좋아요 값은 서버에 저장될 필요가 없는건가요? 
그리구 로직의 경우에도 명확하게 분리시킬 수 있으면 좋아요 
```suggestion
const incrementFavoriteCount = (products, productId) => {
  return products.map((product) =>
    product.id === productId
      ? { ...product, favoriteCount: product.favoriteCount + 1 }
      : product
  );
};

const handleLike = (productId) => {
  setProducts((prevProducts) => incrementFavoriteCount(prevProducts, productId));
};
```

그리고 삼항 연산자가 편하긴 해도 항상 가독성이 좋지만은 않습니다. 따라서 개인 취향에 따라서 이런식으로도 해볼 수 있겠죠 

```
const incrementFavoriteCount = (products, productId) => {
  return products.map((product) => {
    if (product.id === productId) {
      return { ...product, favoriteCount: product.favoriteCount + 1 };
    }
    return product;
  });
};

const handleLike = (productId) => {
  setProducts((prevProducts) => incrementFavoriteCount(prevProducts, productId));
};
```